### ISSUE TITLE
Inter-lens distance not working with iPhone 16

### ISSUE THREAD

**Issue by Somshekardsi** [googlevr/cardboard#486]
The inter-lens distance setting in Cardboard XR does not function on iPhone 16, preventing proper adjustment of lens separation for VR.

Environment:
- Device: iPhone 16
- Unity Version: Unity 6(6000.0.28f1)
- Cardboard XR Version: 1.27.0

Steps to Reproduce:
1. Use the 'Api.SaveDeviceParams' method to set inter-lens distance.
2. Test on iPhone 16.
3. Observe that the inter-lens distance is not applied.

Expected Behavior:
The inter-lens distance should adjust properly for iPhone 16.

Actual Behavior:
The inter-lens distance setting has no effect on iPhone 16.

Additional Context:
The issue does not occur on previous iPhone models.

**Comment by arilow** [comment]
Thanks @Somshekardsi for reaching out to us!

In order to allow us to reproduce this issue, could you please provide:
- URI you are passing to Api.SaveDeviceParams().
- A screenshot where the bug can be depicted.
- Is this reproducible with the Cardboard sample?
- Is this reproducible in other Unity versions?

Thanks!

**Comment by Somshekardsi** [comment]
Thank you for your response!

URI : http://google.com/cardboard/cfg?p=CgZHb29nbGUSEkNhcmRib2FyZCBJL08gMjAxNR0rGBU9JSlcDz0qEAAASEIAAEhCAABIQgAASEJYATUpXA89OggeZnc-Ej6aPlAAYAA
Regarding the second point, could you clarify what kind of screenshot you need? Should it depict the behavior in the app or an error message?
Yes, this issue is reproducible with the Cardboard sample.
Yes, it is also reproducible in other Unity versions.
Let me know if you need additional details!

**Comment by Somshekardsi** [comment]
> Thanks @Somshekardsi for reaching out to us!
> 
> In order to allow us to reproduce this issue, could you please provide:
> 
> * URI you are passing to Api.SaveDeviceParams().
> * A screenshot where the bug can be depicted.
> * Is this reproducible with the Cardboard sample?
> * Is this reproducible in other Unity versions?
> 
> Thanks!

Thank you for your response!

URI : http://google.com/cardboard/cfg?p=CgZHb29nbGUSEkNhcmRib2FyZCBJL08gMjAxNR0rGBU9JSlcDz0qEAAASEIAAEhCAABIQgAASEJYATUpXA89OggeZnc-Ej6aPlAAYAA
Regarding the second point, could you clarify what kind of screenshot you need? Should it depict the behavior in the app or an error message?
Yes, this issue is reproducible with the Cardboard sample.
Yes, it is also reproducible in other Unity versions.
Let me know if you need additional details!

**Comment by arilow** [comment]
Hi! I was referring to a screenshot of the device that depicts the behavior in the app. But if there are any logs that you think should be useful for this, please also share them.

Thanks!

**Comment by Somshekardsi** [comment]
> 1. Api.SaveDeviceParams

Hi! There are no errors after running 'Api.SaveDeviceParams', but the parameters are not applied on iPhone 16. The same code works correctly on previous iPhone versions.

**Comment by ShashankMishra777** [comment]
Hi! I noticed this bug on iPhone 16 pro max.

It was working fine on iPhone 15 and lower versions.

When I upgraded to sdk version 1.2.7, now it is working on my iPhone 16 pro max.

But when running the same version on iPhone 11 (iOS 17), and I go from scene 0 to scene 1 (cardboard scene), it throws me back to scene 0.

**Comment by rjcausarano** [comment]
Hello @ShashankMishra777 and thanks for the update!

I've read your update and it sounds to me like the original inter-lens distance issue that was reported to not work on the iPhone 16 is now solved. Can you please confirm that the issue is solved?

Thanks.

**Comment by ShashankMishra777** [comment]
Yes the issue is solved on iPhone 16 pro max.

But when running the same version on iPhone 11 (iOS 17), and I go from scene 0 to scene 1 (cardboard scene), it throws me back to scene 0. 
I wanted to know if anyone else is facing the issue with lower versions with sdk version 1.2.7

**Comment by rjcausarano** [comment]
@ShashankMishra777 Yes, could you please open another issue? So that we can properly address it separately.
Thanks.

**Comment by rjcausarano** [comment]
@Somshekardsi We’ve observed that after using the 'Api.SaveDeviceParams' method to set inter-lens distance in the Unity HelloCardboard sample app with the following parameters:

- Device: iPhone 16

- Unity Version: Unity 6(6000.0.28f1)

- Cardboard XR Version: 1.27.0

- Uri passed to 'Api.SaveDeviceParams: http://google.com/cardboard/cfg?p=CgZHb29nbGUSEkNhcmRib2FyZCBJL08gMjAxNR0rGBU9JSlcDz0qEAAASEIAAEhCAABIQgAASEJYATUpXA89OggeZnc-Ej6aPlAAYAA

The inter-lens distance was applied just as expected and as observed in previous iPhone devices.
These were the steps taken to attempt to reproduce the issue:
In the [CardboardStartup.cs](https://github.com/googlevr/cardboard-xr-plugin/blob/master/Samples~/hellocardboard-unity/Scripts/CardboardStartup.cs#L52) script, modified the behaviour when the gear button is pressed to apply the new inter-lens distance:
      
 ```
if (Api.IsGearButtonPressed){
  Api.SaveDeviceParams("http://google.com/cardboard/cfg?p=CgZHb29nbGUSEkNhcmRib2FyZCBJL08gMjAxNR0rGBU9JSlcDz0qEAAASEIAAEhCAABIQgAASEJYATUpXA89OggeZnc-Ej6aPlAAYAA");
  Api.ReloadDeviceParams();
}
```

Run the app on the device, you should see something like this:


![Image](https://github.com/user-attachments/assets/dc5b087b-e663-4b25-96bb-a39c79ca9ae3)

After pressing the gear button on the top right corner of the device, the view is:

![Image](https://github.com/user-attachments/assets/486ef9d7-eace-4982-abf4-b7c9066616b9)

Please try these steps on your side and let us know how that goes.

Thanks!
